src/main/java/edu/tseidler/states/Running.java

```diff
@@ -12,11 +14,19 @@ public Running(GameState previousState) {
 
     @Override
     GameState getNextState() {
-        board = new Board(new int[] {5, 15, 3});
-        board.put(new Coordinates(2, 0), BoardField.X);
-        board.put(new Coordinates(0, 5), BoardField.O);
-        System.out.println((board.put(new Coordinates(6, 5), BoardField.O)));
         output.accept(board.draw());
-        return new GameOverState(this);
+        Player currentPlayer = players.getNext();
+        boolean marked = false;
+        int choice = -1;
+        while (!marked && !board.ifFull()) {
+            output.accept(lang.get("PLAYER") + " " + currentPlayer.name + " " + lang.get("NEXT_MOVE") + " " + currentPlayer.mark);
+            choice = InputParser.parsePlayerMarkInput(input.get());
+            marked = board.put(choice, currentPlayer.mark);
```

Comment:
I would prefer make mark field in Player class private and final and create getter